### DISCUSSION TITLE
4 Semester 4 Woche

### DISCUSSION THREAD

**Discussion by MelonigeMelone** [MelonigeMelone/xCalendar#31] [Blogkanal]
# Aktueller Status
Diese Woche hatten wir trotz Feiertag etwas weniger Zeit, konnten trotzdem einen Testplan erstellen, allerdings war die Integration der Test stand jetzt noch nicht möglich, was im Laufe der Woche noch folgen wird.
# Testplan 
Im folgenden geben wir euch einen Überblick über unseren Testplan. Der Upload des detailierten Testplans folgt noch im Laufe der nächsten Tage.

## Umfang der Tests
Der Umfang des Tests wird sich hauptsächlich auf das Backend durch Unittests beziehen. Es gab Überlegungen auch das Frontend zu testen, zum Beispiel mit Hilfe von Selenium, welches Klickpfades simuliert. Dies kann einerseits sehr nützlich sein um Fehler in der Oberfläche zu finden, welche durch zum Beispiel Updates entstehen, allerdings ist der Aufwand dieser Test momentan zu hoch und wir konzentrieren uns auf die Logik das Hauptmerkmal des Backends. 
Als zweitens bei einem Blick auf unsere Risiko Management Tabelle, hatten wir auch Performanz Probleme aufgezählt. Hierzu haben wir bereits Gatling als Idee eingebracht, um unsere Backend Schnittstelle zu testen. Dies soll allerdings lediglich im Falle von deutlichen Performanz Problemen zu Nutze kommen, da auch hier ein enormer Aufwand beim Schreiben solcher Tests entsteht.

## Testabdeckung
Die Test sollen möglichst jede Request die das Backend ausführt testen, also einen Umfang von an die 100% wir haben uns auf 90% festgelegt.

## Testwerkzeug
Als Testwerkzeug nutzen wir JUNIT zur Durchführung.
Zur Ermittlung der Testabdeckung steht noch kein Werkzeug fest.

## Verwaltung der Testfälle
Nach Integration dieser soll jeder Commit mit Jenkins gebaut werden. Während dem Build Prozess sollen alle Tests ausgeführt werden und nur bei erfolgreichem Beendigung aller Testfälle, soll der Commit auch gemergt werden. Gleichzeitig haben wir einen Überblick darüber, ob unsere Testfälle erfolgreich waren bzw. können nachschauen, welche fehlgeschlagen sind. In Jenkins kann man diesen Verlauf hervorragend verfolgen.

**Comment by Charlotte0999** [comment]
Hallo zusammen,
ich finde euren Ansatz gut, ihr scheint euch wirklich Gedanken für die Planung gemacht zu haben und welche Hilfsmittel ihr euch zunutzen macht für die Tests. Ich bin gespannt, wie ihr in Zukunft euer Programm testen werdet. 
Weiterhin viel Erfolg!
